managers/backup-manager/BackupStatusPoller.js

```diff
@@ -0,0 +1,261 @@
+                  metadata: metadata
+                })
+                .tap(updatedResource => logger.debug(`Successfully acquired backup status poller lock for request with options: ${JSON.stringify(options)}\n` +
+                  `Updated resource with poller annotations is: `, updatedResource))
+                .then(() => {
+                  if (!utils.isServiceFabrikOperationFinished(resourceBody.status.state)) {
+                    const instanceInfo = _.chain(response)
+                      .pick('tenant_id', 'backup_guid', 'instance_guid', 'agent_ip', 'service_id', 'plan_id', 'deployment', 'started_at', 'abortStartTime')
+                      .value();
+                    return BackupStatusPoller.checkOperationCompletionStatus(instanceInfo);
+                  }
+                })
+                .catch(AssertionError, err => {
+                  logger.error(`Error occured while polling for backup, marking backup as failed`, err);
+                  return eventmesh.apiServerClient.updateResource({
+                    resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                    resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                    resourceId: metadata.name,
+                    status: {
+                      state: CONST.APISERVER.RESOURCE_STATE.FAILED,
+                      error: utils.buildErrorJson(err)
+                    }
+                  });
+                })
+                .catch(Conflict, () => {
+                  logger.debug(`Not able to acquire backup status poller processing lock for backup with guid ${object.metadata.name}, Request is probably picked by other worker`);
+                });
+            }
+          });
+        });
+    }
+
+    function startPoller(event) {
+      logger.debug('Received Backup Event: ', event);
+      return Promise.try(() => {
+        if ((event.type === CONST.API_SERVER.WATCH_EVENT.ADDED || event.type === CONST.API_SERVER.WATCH_EVENT.MODIFIED) && !BackupStatusPoller.pollers[event.object.metadata.name]) {
+          BackupStatusPoller.pollers[event.object.metadata.name] = true;
+          return poller(event.object)
+            .finally(() => BackupStatusPoller.pollers[event.object.metadata.name] = false);
+        }
+      });
+    }
+    const queryString = `state in (${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS},${CONST.APISERVER.RESOURCE_STATE.ABORTING})`;
+    return eventmesh.apiServerClient.registerWatcher(CONST.APISERVER.RESOURCE_GROUPS.BACKUP, CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP, startPoller, queryString)
+      .then(stream => {
+        logger.debug(`Successfully set watcher on backup resources for backup status polling with query string:`, queryString);
+        return Promise
+          .delay(config.backup.backup_restore_status_check_every)
+          .then(() => {
+            logger.debug(`Refreshing stream after ${config.backup.backup_restore_status_check_every}`);
```

Comment:
Can we update to "Refreshing backup stream"

Reply:
Done